lib/ts/recipe/session/recipeImplementation.ts

```diff
@@ -162,38 +171,59 @@ export default function getRecipeInterface(
             userContext: any;
         }): Promise<Session | undefined> {
             logDebugMessage("getSession: Started");
-            logDebugMessage("getSession: rid in header: " + frontendHasInterceptor(req));
-            logDebugMessage("getSession: request method: " + req.getMethod());
+
+            const sessionOptional =
+                options !== undefined && typeof options !== "boolean" && options.sessionRequired === false;
+            const gotRID = frontendHasInterceptor(req);
+            const method = req.getMethod();
+
+            logDebugMessage("getSession: optional validation: " + sessionOptional);
+            logDebugMessage("getSession: rid in header: " + gotRID);
+            logDebugMessage("getSession: request method: " + method);
             let doAntiCsrfCheck = options !== undefined ? options.antiCsrfCheck : undefined;
-            const transferMethod = config.getTokenTransferMethod({ req, userContext });
+            const preferredTransferMethod = config.getTokenTransferMethod({ req, userContext });
+
+            let transferMethod = preferredTransferMethod === "MISSING_AUTH_HEADER" ? "header" : preferredTransferMethod;
 
-            let accessToken = getToken(config, req, "access", userContext, transferMethod);
+            let accessToken = getToken(req, "access", transferMethod);
             if (accessToken === undefined) {
+                const fallbackMethod = transferMethod === "cookie" ? "header" : "cookie";
                 // We are checking if we could've gone ahead with validation if the transferMethod was different
                 // However, we don't want to do the fallback here, but force a call to refresh
                 // This is done to ensure that the browsers update the transfermethod in a timely manner (basically on the next API call)
                 // instead of waiting for the session to expire
```

Comment:
this comment is misleading. You are actually allowing fallback method in case of MISSING_AUTH_HEADER, but the comment says that "However, we don't want to do the fallback here, but force a call to refresh"